src/resources/glossary.md

```diff
@@ -0,0 +1,19 @@
+---
+title: Glossary
+description: A glossary reference for terminology used across dart.dev
+sitemap: false
+---
+
+The following are definitions of terminology used across the Dart documentation.
+
+## Irrefutable patterns
+
+_Irrefutable patterns_ are patterns that always match. 
+Irrefutable patterns appear in _irrefutable contexts_: 
+_declaration context_ and _assignment context_.
+
+## Refutable patterns
+
+A _refutable pattern_ is a pattern that can be tested against a value to
+determine if the pattern matches the value. If not, the pattern _refutes_ the value.
+Refutable patterns appear in _matching contexts_.
```

Comment:
Do we want to enumerate the matching contexts?

Do we want to provide examples of a subset of such contexts (both here and for irrefutable contexts)?

Reply:
I was thinking maybe "Patter contexts" could be a glossary section too, or maybe each individual pattern context. That kind of section would be nice to enumerate and provide examples. I'll make  a note with the ret of my patterns plan to return to this.